components/tyk-mdcb/templates/deployment-mdcb.yaml

```diff
@@ -0,0 +1,144 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mdcb-{{ include "tyk-mdcb.fullname" . }}
+  labels:
+    {{- include "tyk-mdcb.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.mdcb.autoscaling.enabled }}
+  replicas: {{ .Values.mdcb.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "tyk-mdcb.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.mdcb.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "tyk-mdcb.labels" . | nindent 8 }}
+        {{- with .Values.mdcb.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- with .Values.mdcb.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "tyk-mdcb.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.mdcb.podSecurityContext | nindent 8 }}
+      containers:
+        - name: mdcb-{{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.mdcb.containerSecurityContext | nindent 12 }}
+          image: "{{ .Values.mdcb.image.repository }}:{{ .Values.mdcb.image.tag }}"
+          imagePullPolicy: {{ .Values.mdcb.image.pullPolicy }}
+          command: ["/opt/tyk-sink/tyk-sink", "--c=/etc/tyk-sink/tyk_sink.conf"]
+          ports:
+            - containerPort: {{ .Values.mdcb.listenPort }}
+            - containerPort: {{ .Values.mdcb.probes.healthCheckPort }}
+          livenessProbe:
+            httpGet:
+              path: {{ .Values.mdcb.probes.liveness.path }}
+              port: {{ .Values.mdcb.probes.healthCheckPort}}
+              scheme: {{ .Values.mdcb.probes.liveness.scheme }}
```

Comment:
We can and i do not think that users will ever need to configure scheme. I just made it configurable to have consistency in probes. but if you think it will create confusion, I am okay to refactor this piece

Reply:
I think we should configure it ourselves.

Reply:
updated